front/components/assistant/conversation/Conversation.tsx

```diff
@@ -8,11 +10,28 @@ export default function Conversation({
   conversationId: string;
   owner: WorkspaceType;
 }) {
-  const { conversation, isConversationError, isConversationLoading } =
-    useConversation({
-      conversationId,
-      workspaceId: owner.sId,
-    });
+  const {
+    conversation,
+    isConversationError,
+    isConversationLoading,
+    mutateConversation,
+  } = useConversation({
+    conversationId,
+    workspaceId: owner.sId,
+  });
+
+  useEffect(() => {
+    const es = new EventSource(
+      `/api/w/${owner.sId}/assistant/conversations/${conversationId}/events`
+    );
+    es.onmessage = () => {
+      void mutateConversation();
+    };
```

Comment:
What happens on time out? We need to handle reconnection no?

Reply:
Just seeing that. I added re-connection yes.